lib/payment_processor/braintree/webhook_handler.rb

```diff
@@ -51,8 +51,10 @@ def process_notification
         when 'subscription_charged_unsuccessfully'
           handle_failed_subscription_charge
         else
-          Rails.logger.info("Unsupported Braintree::WebhookNotification received of type '#{@notification.kind}'")
+          Rails.logger.info("Unsupported Braintree::WebhookNotification received of type '#{notification.kind}'")
         end
+      rescue
+        Rails.logger.error("Braintree webhook handling failed for '#{notification.kind}', for subscription ID '#{notification.subscription.id}'")
```

Comment:
@Tuuleh I think this is not a good idea. Having a "rescue all" clause is usually not recommended. I'd recommend having specific rescues for the kind of errors you're expecting in each handler method.